cylc/flow/task_pool.py

```diff
@@ -318,73 +318,45 @@ def compute_runahead(self, force=False) -> bool:
             ilimit = int(limit)  # type: ignore
             count_cycles = True
 
-        base_point: 'PointBase'
-        points: List['PointBase'] = []
+        base_point: Optional['PointBase'] = None
 
+        # First get the runahead base point.
         if not self.main_pool:
-            # No tasks yet, just consider sequence points.
-            if count_cycles:
-                # Get the first ilimit points in each sequence.
-                # (After workflow start point - sequence may begin earlier).
-                points = [
-                    point
-                    for plist in [
-                        seq.get_first_n_points(
-                            ilimit, self.config.start_point)
-                        for seq in self.config.sequences
-                    ]
-                    for point in plist
-                ]
-                # Drop points beyond the limit.
-                points = sorted(points)[:ilimit + 1]
-                base_point = min(points)
-
-            else:
-                # Start at first point in each sequence.
-                # (After workflow start point - sequence may begin earlier).
-                points = [
-                    point
-                    for point in {
-                        seq.get_first_point(self.config.start_point)
-                        for seq in self.config.sequences
-                    }
-                    if point is not None
-                ]
-                base_point = min(points)
-                # Drop points beyond the limit.
-                points = [
-                    point
-                    for point in points
-                    if point <= base_point + limit
-                ]
-
+            # Find the earliest sequence point beyond the workflow start point.
+            points = [
+                point
+                for point in {
+                    seq.get_first_point(self.config.start_point)
+                    for seq in self.config.sequences
+                }
+                if point is not None
+            ]
+            base_point = min(points)
```

Comment:
Look like the method only requires the `points` list to compute the minimum so no need to construct it:

```suggestion
            base_point = min(
                point
                for point in {
                    seq.get_first_point(self.config.start_point)
                    for seq in self.config.sequences
                }
                if point is not None
            )
```